modules/workflow-service-api/src/main/java/org/opencastproject/workflow/api/WorkflowServiceDatabaseImpl.java

```diff
@@ -197,6 +197,56 @@ public long countWorkflows(WorkflowInstance.WorkflowState state) throws Workflow
     }
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * @see WorkflowServiceDatabase#getLatestWorkflowInstances(int limit, long token)
+   */
+  public List<WorkflowInstance> getLatestWorkflowInstances(int limit, long token) throws WorkflowDatabaseException {
+
+    EntityManager em = null;
+    try {
+      em = emf.createEntityManager();
+
+      var query = em.createNamedQuery("Workflow.findLatest", WorkflowInstance.class);
+
+      String orgId = securityService.getOrganization().getId();
+      query.setParameter("organizationId", orgId);
+      query.setParameter("startToken", token);
+      query.setMaxResults(limit);
+      logger.debug("Requesting latest workflows using query: {}", query);
+      return query.getResultList();
+    } catch (Exception e) {
+      throw new WorkflowDatabaseException(e);
+    } finally {
+      if (em != null)
+        em.close();
+    }
+  }
+
+    /**
+   * {@inheritDoc}
+   *
+   * @see WorkflowServiceDatabase#countMediaPackages()
+   */
+  public int countMediaPackages() throws WorkflowDatabaseException {
+
+    EntityManager em = null;
+    try {
+      em = emf.createEntityManager();
+
+      var query = em.createNamedQuery("Workflow.countLatest", Long.class);
+      logger.debug("Counting latest workflows using query: {}", query);
+      final Number countResult = query.getSingleResult();
+      return countResult.intValue();
```

Comment:
When I suggested using a `long` I actually meant it to simplify the whole method:

```suggestion
  public long countMediaPackages() throws WorkflowDatabaseException {
    EntityManager em = null;
    try {
      em = emf.createEntityManager();

      var query = em.createNamedQuery("Workflow.countLatest", Long.class);
      logger.debug("Counting latest workflows using query: {}", query);
      return query.getSingleResult();
```

Reply:
Missed that I still had a comment on this. It's not really important, though. I'll merge it like this.